lib/string_reverse.rb

```diff
@@ -1,4 +1,18 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+  # checks to see if there is a string to reverse
+  return nil if my_string == nil || my_string.length == 0
+
+  string = 0 # first character of string

```

Comment:
Minor: Technically, this will be "the index of the first character of string" and not "first character of string". :-)